build.gradle

```diff
@@ -893,6 +893,10 @@ project(':core') {
       implementation libs.dropwizardMetrics
       exclude module: 'slf4j-log4j12'
       exclude module: 'log4j'
+      // ZooKeeper moved from log4j to logback in v3.8.0,
+      // but Kafka relies on log4j. We are removing this additional dependency.
```

Comment:
> but Kafka relies on log4j

this is not correct. Kafka uses slf4j APIs with reload4j as the logging framework. I would suggest to rephrase this as "Zookeeper uses logback as the logging framework but Kafka uses reload4j, hence we want to avoid having multiple frameworks in dependency chain, so removing logback".

Next, can you check why Zk logs are working even if we are using reload4j at runtime? (the answer might be that Zk uses slf4j and underlying framework is pluggable)

Reply:
Yup, this is the exact reason. Kafka uses slf4j + reload4j. Zookeeper uses slf4j + logback. When we remove logback we fallback to reload4j for both Zookeeper and Kafka. I will update the comment.

Reply:
Thank you. That clarifies things. This is a safe change now.